src/guiguts/tools/jeebies.py

```diff
@@ -0,0 +1,297 @@
+                    # Get whole-words slice from original para text that is approximately
+                    # centered on the 3-form phrase.
+                    _, _, centered_slice = self.get_para_slice(
+                        match_obj.start(0), wbs[rec_num]
+                    )
+                    self.add_to_dialog(header, centered_slice, checker_dialog)
+
+        return suspects_count
+
+    def get_para_slice(self, indx: int, para: str) -> tuple[int, int, str]:
+        """Define the start and end positions of a slice of a string containing
+           only whole words and also return that slice.
+
+        Args:
+            indx: normally the position in string about which the slice will be
+                  centered. If the required slice is the head or tail of the string
+                  then indx will be 0 or len(para) - 1 respectively.
+            para: a text string from which a 'whole words' slice will be extracted.
+        """
+
+        # The slice of the paragraph text to be delimited will
+        # have as many whole words that can fit into about 60
+        # characters.
+
+        if indx == 0:
+            # Required slice is the head of the paragraph string.
+            llim = 0
+            rlim = 60
+
+            rlim = min(rlim, len(para))
+
+        elif indx == len(para) - 1:
+            # Required slice is the tail of the paragraph string.
+            rlim = len(para)
+            llim = rlim - 60
+
+            llim = max(llim, 0)
+
+        else:
+            # Required slice is to be centered around indx if possible.
+            llim = indx - 30
+            rlim = indx + 30
+
+            if llim < 0:
+                llim = 0
+                rlim = 60
+
+            if rlim > len(para):
+                llim = len(para) - 60
+                rlim = len(para)
+
+            llim = max(llim, 0)
+
+        # As a result of the abitrary width limit chosen above, the start
+        # and end of the slice of the paragrpah string defined by llim and
+        # rlim may have partial words. Expand the slice at both ends until first
+        # whitespace character found or start or end of paragraph encountered.
+        # The slice that is defined will now have only whole words.
+
+        while llim > 0 and para[llim : llim + 1] != " ":
+            llim -= 1
+
+        # If not at left margin set llim to start of first word in slice.
+        if llim != 0:
+            llim += 1
+
+        while rlim < len(para) and para[rlim : rlim + 1] != " ":
+            rlim += 1
+
+        # If not at right margin set rlim to end of last word in slice.
+        if rlim != len(para) - 1:
+            rlim -= 1
+
+        return llim, rlim + 1, para[llim : rlim + 1]
+
+    def find_in_dictionary(self, three_form: str) -> int:
+        """returns frequency in hebe phrase corpus or zero"""
+
+        words = three_form.split()
+        key = words[0] + "|" + words[1] + "|" + words[2]
```

Comment:
Minor suggestion: if `words` only has 3 things in it you could do `"|".join(words)`